src/routes/RemediationDetails.js

```diff
@@ -170,6 +179,27 @@ const RemediationDetails = ({
 
   const { status, remediation } = selectedRemediation;
 
+  const fetchDisabledValue = useCallback(async () => {
+    return Promise.resolve(
+      instance.get(
+        `${API_BASE}/remediations/${remediation.id}/connection_status`
+      )
+    ).then(({ data }) => {
+      setIsDisabled(
+        !context.permissions.execute ||
+          !executable ||
+          isFedramp ||
+          data[0].connection_status !== 'connected'
+      );
+    });
+  });
```

Comment:
This should have `context`, `executable` and `isFedramp` as dependencies. 

It might be good to extract the fetching into a hook. something like `useConnectionStatus` that returns only the result of `data[0].connection_status !== 'connected'`

Reply:
Extracting it into a hook will also make it easier to add a test for. (😉)

Reply:
Originally I did this, but I had some issues where the other variables would affect things in a weird way, and even if I returned true (for disabling it), it wouldnt update the button. The new changes update everything to be a hook, but it returns a isDisabled value and we pass in those other variables as params

Reply:
The hook looks good to me, but if you want we could take a look together to make it purely concerned about the connection status fetching.